test/tests_basic.jl

```diff
@@ -192,6 +235,24 @@ end
     @test_nowarn p4est_destroy(p4est)
     @test_nowarn p4est_connectivity_destroy(connectivity)
   end
+
+  # This test is based on init_neighbor_rank_connectivity_iter_face_inner from Trixi.jl
+  # See https://github.com/trixi-framework/Trixi.jl/blob/main/src/solvers/dgsem_p4est/dg_parallel.jl
+  @testset "nested attributes" begin
+    iter_face_nested_attributes_c = @cfunction(iter_face_nested_attributes, Cvoid,
+                                               (Ptr{p4est_iter_face_info_t}, Ptr{Cvoid}))
+    connectivity = @test_nowarn p4est_connectivity_new_brick(2, 2, 0, 0)
+    p4est = @test_nowarn p4est_new_ext(MPI.COMM_WORLD, connectivity, 0, 0, true, 0, C_NULL, C_NULL)
```

Comment:
Could you please include some local refinements here so that you also get non-conforming interfaces (with hanging nodes)?

Reply:
You ca find some example how to get this here: https://github.com/trixi-framework/Trixi.jl/blob/main/examples/p4est_2d_dgsem/elixir_advection_nonconforming_flag.jl

Reply:
Yes, I will do that.

Reply:
Please ping me for a review when you are ready (or have questions along the way)

Reply:
@ranocha I included a local refinement so that there are also hanging nodes. Now, all quadrants are locally available (`sides[full_side].is.full.is_ghost == false && all(sides[hanging_side].is.hanging.is_ghost .== false)` returns `true`). Should I also adapt the test such that this is not the case?

Reply:
No, I think it's fine